nilearn/surface/surface.py

```diff
@@ -512,6 +514,50 @@ def vol_to_surf(img, surf_mesh,
     return texture.T
 
 
+def _load_surf_files_gifti_gzip(surf_file):
+    """Load surface data Gifti files which are gzipped. This
+    function is used by load_surf_mesh and load_surf_data for
+    extracting gzipped files.
+
+    Part of the code can be removed while bumping nibabel 2.0.2
+    """
+    with gzip.open(surf_file) as f:
+        as_bytes = f.read()
+    if LooseVersion(nibabel.__version__) >= LooseVersion('2.1.0'):
+        parser = gifti.GiftiImage.parser()
+        parser.parse(as_bytes)
+        gifti_img = parser.img
+    else:
+        from nibabel.gifti.parse_gifti_fast import ParserCreate, Outputter
+        parser = ParserCreate()
+        parser.buffer_text = True
+        out = Outputter()
+        parser.StartElementHandler = out.StartElementHandler
+        parser.EndElementHandler = out.EndElementHandler
+        parser.CharacterDataHandler = out.CharacterDataHandler
+        parser.Parse(as_bytes)
+        gifti_img = out.img
+    return gifti_img
+
+
+def _load_surf_depth_img_to_data(gifti_img):
+    """Load surface sulcal depth image in nibabel.gifti.GiftiImage
+    to data
+
+    Used by load_surf_data function in common to surface sulcal data
+    acceptable to .gii or .gii.gz
+    """
+    try:
```

Comment:
maybe this can be simplified a bit e.g.
```
    if not gifti_img.darrays:
        raise ValueError('Gifti must contain at least one data array')
    return np.asarray([arr.data for arr in gifti_img.darrays]).T.squeeze()
```